src/pprof_export.jl

```diff
@@ -0,0 +1,266 @@
+
+function iterate_summary_result(
+    parent_name,
+    top_level_field_result::FieldResult,
+)
+    return Channel() do ch
+        parent_path = parent_name
+        parent_field = top_level_field_result
+        key = String[]
+
+        for (name,field) in parent_field.children
+            push!(key, name)
+            put!(ch,
+                MemoryInspector._path_str(parent_path, parent_field, name) =>
+                    field)
+        end
+    end
+end
+
+struct FieldTraversalNode
+    path::String
+    parent_field::Union{Nothing,FieldResult}
+    name::String
+    field::FieldResult
+end
+
+"""
+    _enter!(dict::OrderedDict{T, Int64}, key::T) where T
+
+Resolves from `key` to the index (zero-based) in the dict.
+Useful for the Strings table
+
+NOTE: We must use Int64 throughout this package (regardless of system word-size) b/c the
+proto file specifies 64-bit integers.
+"""
+function _enter!(dict::OrderedDict{T, Int64}, key::T) where T
+    if haskey(dict, key)
+        return dict[key]
+    else
+        l = Int64(length(dict))
+        dict[key] = l
+        return l
+    end
+end
+
+using Base.StackTraces: StackFrame
+
+# TODO:
+# - Mappings
+
+macro pprof(exs...)
+    args = exs[1:end-1]
+    e = exs[end]
+    :(_pprof($(string(e)), $(esc(e)); $((esc(a) for a in args)...)))
+end
+
+"""
+    _pprof(name, obj;
+            web = true, webhost = "localhost", webport = 57599,
+            out = "profile.pb.gz", from_c = true, drop_frames = "", keep_frames = "",
+            ui_relative_percentages = true,
+         )
+
+Fetches the collected `Profile` data, exports to the `pprof` format, and (optionally) opens
+a `pprof` web-server for interactively viewing the results.
+
+If `web=true`, the web-server is opened in the background. Re-running `pprof()` will refresh
+the web-server to use the new output.
+
+If you manually edit the output file, `PProf.refresh()` will refresh the server without
+overwriting the output file. `PProf.kill()` will kill the server.
+
+# Arguments:
+- `data::Vector{UInt}`: The data provided by `Profile.fetch` [optional].
+
+# Keyword Arguments
+- `web::Bool`: Whether to launch the `go tool pprof` interactive webserver for viewing results.
+- `webhost::AbstractString`: If using `web`, which host to launch the webserver on.
+- `webport::Integer`: If using `web`, which port to launch the webserver on.
+- `out::String`: Filename for output.
+- `from_c::Bool`: If `false`, exclude frames that come from from_c. Defaults to `true`.
+- `drop_frames`: frames with function_name fully matching regexp string will be dropped from the samples,
+                 along with their successors.
+- `keep_frames`: frames with function_name fully matching regexp string will be kept, even if it matches drop_functions.
+- `ui_relative_percentages`: Passes `-relative_percentages` to pprof. Causes nodes
+  ignored/hidden through the web UI to be ignored from totals when computing percentages.
+"""
```

Comment:
TODO: This docstring needs to be updated or removed. This was just copy/pasted from the PProf.jl repo.

Probably instead i should put the docstring on the macro.